cmd/delete/cluster.go

```diff
@@ -15,68 +15,85 @@ import (
+	"github.com/sighupio/furyctl/internal/distribution"
 	cobrax "github.com/sighupio/furyctl/internal/x/cobra"
 	execx "github.com/sighupio/furyctl/internal/x/exec"
-	yamlx "github.com/sighupio/furyctl/internal/x/yaml"
+	netx "github.com/sighupio/furyctl/internal/x/net"
 )
 
-var (
-	ErrYamlUnmarshalFile = errors.New("error unmarshaling yaml file")
-	ErrDebugFlagNotSet   = errors.New("debug flag not set")
-	ErrFuryctlFlagNotSet = errors.New("furyctl flag not set")
-	ErrPhaseFlagNotSet   = errors.New("phase flag not set")
-	ErrDryRunFlagNotSet  = errors.New("dry-run flag not set")
-	ErrForceFlagNotSet   = errors.New("force flag not set")
-)
+var ErrParsingFlag = errors.New("error while parsing flag")
 
-func NewClusterCmd() *cobra.Command {
+func NewClusterCmd(version string) *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "cluster",
 		Short: "Deletes a cluster",
 		RunE: func(cmd *cobra.Command, args []string) error {
 			debug, ok := cobrax.Flag[bool](cmd, "debug").(bool)
 			if !ok {
-				return ErrDebugFlagNotSet
+				return fmt.Errorf("%w: debug", ErrParsingFlag)
 			}
 
 			furyctlPath, ok := cobrax.Flag[string](cmd, "config").(string)
 			if !ok {
-				return ErrFuryctlFlagNotSet
+				return fmt.Errorf("%w: config", ErrParsingFlag)
+			}
+
+			distroLocation, ok := cobrax.Flag[string](cmd, "distro-location").(string)
+			if !ok {
+				return fmt.Errorf("%w: distro-location", ErrParsingFlag)
 			}
 
 			phase, ok := cobrax.Flag[string](cmd, "phase").(string)
 			if !ok {
-				return ErrPhaseFlagNotSet
+				return fmt.Errorf("%w: phase", ErrParsingFlag)
 			}
 
 			dryRun, ok := cobrax.Flag[bool](cmd, "dry-run").(bool)
 			if !ok {
-				return ErrDryRunFlagNotSet
+				return fmt.Errorf("%w: dry-run", ErrParsingFlag)
 			}
 
 			force, ok := cobrax.Flag[bool](cmd, "force").(bool)
 			if !ok {
-				return ErrForceFlagNotSet
+				return fmt.Errorf("%w: debug", ErrParsingFlag)
 			}
 
-			execx.Debug = debug
-
 			// Init paths.
 			homeDir, err := os.UserHomeDir()
 			if err != nil {
 				return fmt.Errorf("error while getting user home directory: %w", err)
 			}
 
-			minimalConf, err := yamlx.FromFileV3[config.Furyctl](furyctlPath)
+			// Init first half of collaborators.
+			client := netx.NewGoGetterClient()
+			executor := execx.NewStdExecutor()
+			distrodl := distribution.NewDownloader(client, debug)
+
+			execx.Debug = debug
+
+			// Download the distribution.
+			logrus.Info("Downloading distribution...")
+			res, err := distrodl.Download(version, distroLocation, furyctlPath)
 			if err != nil {
-				return fmt.Errorf("%w: %s", ErrYamlUnmarshalFile, err)
+				return fmt.Errorf("error while downloading distribution: %w", err)
 			}
 
-			basePath := filepath.Join(homeDir, ".furyctl", minimalConf.Metadata.Name)
+			basePath := filepath.Join(homeDir, ".furyctl", res.MinimalConf.Metadata.Name)
 
-			clusterDeleter, err := cluster.NewDeleter(minimalConf, phase, basePath)
+			binPath := filepath.Join(homeDir, ".furyctl", "bin")
```

Comment:
shouldn't this be configurable via a flag? the validate dependency command handles that

Reply:
true, as is bin-path is only used in validate dependencies. Now we can add bin-path to every command where binaries are needed, or remove bin-path from validation cmd

Reply:
after talking with @ralgozino we will add bin-path flag to every command where binaries are needed